core/wiki/macros/tree.tid

```diff
@@ -1,53 +1,62 @@
 title: $:/core/macros/tree
 tags: $:/tags/Macro
 
-\define leaf-node(prefix)
+\define link(full-title, chunk)
+<$link to=<<__full-title__>>>
+<$text text=<<__chunk__>>/>
+</$link>
+\end
+
+\define leaf-node(prefix, chunk)
 <li>
-<$list filter="""[[$prefix$$(chunk)$]is[shadow]] [[$prefix$$(chunk)$]is[tiddler]] +[removeprefix[$prefix$]] +[limit[1]]""" 
-emptyMessage="""<$text text="$prefix$$(chunk)$"/>""">
-<span>{{$:/core/images/file}}</span> <$link to="""$prefix$$(chunk)$""">
-<$view field="title"/>
-</$link> 
+<$list filter="""[[$prefix$$chunk$]is[shadow]] [[$prefix$$chunk$]is[tiddler]]""" variable="full-title">
```

Comment:
I believe here we should use `<$list filter="[<__prefix__>addsuffix<__chunk__>is[shadow]] ...>`

Reply:
@morosanuae sorry, I've spotted another thing:

here you've removed `+[removeprefix[$prefix$]] +[limit[1]]` and the `emptyMessage`

they are for cases where we have a tiddler like `$:/state/something/something-else/another-thing` but there are no tiddlers `$:/state/something` or `$:/state/something/something-else`

removing the check and the emptyMessage causes `$:/state/something/something-else/another-thing` not to be displayed I think

Reply:
the same below where the other emptyMessage was

Reply:
I will do some tests. Thanks

Reply:
yes, I may be totally wrong here

Reply:
I gave some thought about it and it looks like this situation is not possible in my opinion.

Reply:
The code suffered some extensive modifications throughout the time and the overall logic changed. The redundant code is from @Jermolene initial version of the macro.

Reply:
Anyway if you spot something else please tell me because I want to publish the new version soon. Thanks.

Reply:
thanks, I see :+1:

Reply:
>Anyway if you spot something else please tell me because I want to publish the new version soon. Thanks.

I'm done spotting

Reply:
:)